### PR TITLE
DCS-432 Upgrading libraries

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: job/gatherActiveLdus.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ const getBookingIds = async () => {
   }
 
   const { rows } = await db.query(query)
-  return rows.map(row => row.booking_id)
+  return rows.map((row) => row.booking_id)
```
Comment (andrewrlee) [inline job/gatherActiveLdus.js:24]: latest prettier adds brackets to single args which accounts for 700 changes across the files

File: log.js [hunk 2]
```diff
@@ -13,57 +13,36 @@ winston.addColors({
...
-    timestamp: true,
-    stringify: true,
-    handleExceptions: true,
+const flattened = winston.format((meta) => {
+  const { message, level, timestamp: ts, ...rest } = meta
+  const result = flattenMeta(rest)
+  Object.entries(result).forEach(([key, val]) => {
+    // eslint-disable-next-line no-param-reassign
+    meta[key] = val
   })
```
Comment (andrewrlee) [inline log.js:22]: this is really weird - for some reason you return anything that isn't the original meta object or it breaks logging entirely. Even if I create an object with the exact same state..

File: log.js [hunk 3]
```diff
@@ -13,57 +13,36 @@ winston.addColors({
...
+
+const logger = winston.createLogger({
+  format,
+  transports: [new winston.transports.Console({ level: 'info', handleExceptions: true })],
+})
 
 if (appInsights) {
   logger.info('Activating application insights logger')
 
   logger.add(
```
Comment (andrewrlee) [inline log.js:39]: Definitely need to be careful around testing logging with this change as the api changed significantly and support for application insights seems a bit precarious.

File: package.json [hunk 4]
```diff
@@ -28,15 +28,12 @@
...
-    },
-    "ignore": [
-      "package.json"
+    "*.{js,css}": [
+      "prettier --write",
+      "eslint --fix"
+    ],
+    "*.{json}": [
+      "prettier --write"
     ]
```
Comment (andrewrlee) [inline package.json:37]: lint staged config options have changed

File: server/app.js [hunk 5]
```diff
@@ -12,40 +12,42 @@ const flash = require('connect-flash')
 const pdfRenderer = require('@ministryofjustice/express-template-to-pdf')
 
 const helmet = require('helmet')
+const noCache = require('nocache')
```
Comment (andrewrlee) [inline server/app.js:15]: noCache has been deprecated in helmet and extracted into its own package

File: server/app.js [hunk 6]
```diff
@@ -12,40 +12,42 @@ const flash = require('connect-flash')
...
 const ensureHttps = require('./utils/ensureHttps')
 
-const config = require('../server/config')
+const config = require('./config')
 const healthFactory = require('./services/healthcheck')
 
 const logger = require('../log.js')
 const auth = require('./authentication/auth')
 
-const defaultRouter = require('../server/routes/default')
```
Comment (andrewrlee) [inline server/app.js:27]: eslint now warns about paths which aren't their optimal

File: server/services/healthcheck.js [hunk 7]
```diff
@@ -4,11 +4,11 @@ const service = (name, url) => {
...
   return () =>
     check()
-      .then(result => ({ name, status: 'UP', message: result }))
-      .catch(err => ({ name, status: 'ERROR', message: err }))
+      .then((result) => ({ name, status: 'UP', message: result }))
+      .catch((err) => ({ name, status: 'ERROR', message: err }))
 }
 
-const gatherCheckInfo = (total, currentValue) => Object.assign({}, total, { [currentValue.name]: currentValue.message })
+const gatherCheckInfo = (total, currentValue) => ({ ...total, [currentValue.name]: currentValue.message })
```
Comment (andrewrlee) [inline server/services/healthcheck.js:11]: this got automatically fixed by eslint

File: server/services/notifications/caAndDmNotificationSender.js [hunk 8]
```diff
@@ -112,28 +112,19 @@ module.exports = function createCaAndDmNotificationSender(
...
-    return mailboxes.map(mailbox => {
+    return mailboxes.map((mailbox) => {
       return { personalisation: { ...common, dm_name: mailbox.name }, email: mailbox.email, templateName: 'STANDARD' }
     })
   }
 
   const notificationDataMethod = {
     CA_RETURN: R.compose(
-      R.then(replaceTemplateNames({ STANDARD: 'CA_RETURN', COPY: 'CA_RETURN_COPY' })),
+      R.andThen(replaceTemplateNames({ STANDARD: 'CA_RETURN', COPY: 'CA_RETURN_COPY' })),
```
Comment (andrewrlee) [inline server/services/notifications/caAndDmNotificationSender.js:122]: changes to ramda api - this clashed with the concept of thenables so they had to rename the method.

File: job/gatherActiveLdus.js (hunk 1 above)
Comment (bell-pa) [reply job/gatherActiveLdus.js:24]: Is that an improvement? Can it be turned off?  Perhaps its best to accept that change now that you've implemented it!

File: server/services/notifications/caAndDmNotificationSender.js [hunk 9]
```diff
@@ -14,7 +14,7 @@ module.exports = function createCaAndDmNotificationSender(
   configClient,
   notificationSender,
   nomisClientBuilder,
-  { notifications: { activeNotificationTypes, clearingOfficeEmail, clearingOfficeEmailEnabled }, domain }
+  { notifications: { activeNotificationTypes, clearingOfficeEmail, clearingOfficeEmailEnabled, ..._ }, domain, ...rest }
```
Comment (brightonsbox) [inline server/services/notifications/caAndDmNotificationSender.js:17]: What's happening here with this change?

**Pull request merged**